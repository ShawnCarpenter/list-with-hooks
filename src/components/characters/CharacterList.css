.CharacterList {
  ul {
    display: flex;
    flex-wrap: wrap;
  }
  li {
    list-style-type: none;
    border: .2rem solid #272727;
    border-radius: .3rem;
    margin: .2rem;
    padding: .2rem;
  }
}
