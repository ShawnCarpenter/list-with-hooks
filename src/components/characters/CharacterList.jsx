import React from 'react';
import PropTypes, { shape } from 'prop-types';
import Character from './Character';
import styles from './CharacterList.css';

const CharacterList = ({ characters }) => {
  
  return (
    <div className={styles.CharacterList}>
      <ul>{
        characters.map(character => (
          <li key={character.name}>
            <Character
              name={character.name}
              image={character.image} 
            />
          </li>
        ))
      }</ul>
    </div>
      
  );
};

CharacterList.propTypes = {
  characters:PropTypes.arrayOf(shape({ 
    name:PropTypes.string,  }))
};

export default CharacterList;
