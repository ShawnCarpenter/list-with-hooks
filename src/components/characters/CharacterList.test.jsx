import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';


describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <CharacterList characters={[
        { name:'Bob', image:'http://image.com/pic.jpg' }]}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
