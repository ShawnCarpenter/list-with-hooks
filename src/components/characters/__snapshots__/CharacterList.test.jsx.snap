// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <div
    class="CharacterList"
  >
    <ul>
      <li>
        <div
          class="Character"
        >
          <a
            href="detail/Bob"
          >
            <h3>
              Bob
            </h3>
            <img
              alt="Bob"
              src="http://image.com/pic.jpg"
            />
          </a>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
